sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -162,6 +165,8 @@ class SQLBuilder private (
     case p: Except =>
       build("(" + toSQL(p.left), ") EXCEPT (", toSQL(p.right) + ")")
 
+    case p @ SubqueryAlias(_, Limit(_, _), _) => build(toSQL(p.child), "AS", p.alias)
```

Comment:
double `((` `))` is no harm, we should prefer the simplicity of the generator code over the generated SQL.